homework-classes/App.js

```diff
@@ -1,78 +1,100 @@
 'use strict';
 
 /* global Util, Repository, Contributor */
+const root = document.getElementById('root');
+const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
 
 class App {
   constructor(url) {
-    this.mainContainer = null;
-    this.initialize(url);
+    this.createHeader(url);
+  }
+
+  createHeader(url) {
+    const header = Util.createAndAppend('header', root, {
+      class: 'container jumbotron mt-3 mb-3 px-5',
+    });
+    Util.createAndAppend('main', root, {
+      id: 'container',
+      class: 'container align-items-start',
+    });
+    const headerDiv = Util.createAndAppend('div', header, {
+      class: 'header-group d-flex flex-row justify-content-between mb-5 px-5',
+    });
+    Util.createAndAppend('h2', headerDiv, {
+      text: 'Hack Your Future Repositories',
+      class: 'display-4 d-flex align-items-center mx-auto col-sm header-title',
+    });
+    const hyfLink = Util.createAndAppend('a', headerDiv, {
+      href: 'https://www.hackyourfuture.net/',
+      target: '_blank',
+    });
+    Util.createAndAppend('img', hyfLink, {
+      class: 'img-fluid rounded img-thumbnail mx-auto',
+      src: './hyf.png',
+      alt: 'hack your future thumbnail',
+    });
+    Util.createAndAppend('select', header, {
+      id: 'repo-select',
+      class: 'form-control',
+      'aria-label': 'Hack Your Future Repositories Selection',
+    });
+    this.fetchRepositories(url);
   }
 
   /**
-   * Initialization
    * @param {string} url The GitHub URL for obtaining the organization's repositories.
    */
-  async initialize(url) {
```

Comment:
The idea of the `.initialization()` is tell the overall story about what should at startup. For instance:

```js
class App {
  constructor(url) {
    this.initialize(url);
  }

  async initialize(url) {
    this.root = document.getElementById('root');
    this.createHeader();
    const repos = await Util.fetchJSON(url);
    this.repos = repos
      .map(repo => new Repository(repo))
      .sort((a, b) => a.name().localeCompare(b.name()));
    this.appendRepositoriesToSelect();
  }
  ...
}

You can save values that you need often in the class, such as the root element as properties in the class (using `this`), instead of using global variables. Other candidates for this are the `select` element and the container elements for the repositories and contributors. In that way, you do not need to use `getElementById` each time you need a reference.
```